internal/auth/credentials.go

```diff
@@ -0,0 +1,629 @@
+// Copyright © 2025 Ping Identity Corporation
+
+package auth_internal
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/pingidentity/pingcli/internal/configuration/options"
+	"github.com/pingidentity/pingcli/internal/errs"
+	"github.com/pingidentity/pingcli/internal/profiles"
+	"github.com/pingidentity/pingone-go-client/config"
+	svcOAuth2 "github.com/pingidentity/pingone-go-client/oauth2"
+	"golang.org/x/oauth2"
+)
+
+// No init() function needed - direct integration is cleaner
+
+// Token storage keys for different authentication methods
+const (
+	deviceCodeTokenKey        = "device-code-token"
+	authCodeTokenKey          = "auth-code-token" // #nosec G101 -- This is a keychain identifier, not a credential
+	clientCredentialsTokenKey = "client-credentials-token"
+)
+
+// getTokenStorage returns the appropriate keychain storage instance for the given authentication method
+func getTokenStorage(authMethod string) *svcOAuth2.KeychainStorage {
+	return svcOAuth2.NewKeychainStorage("pingcli", authMethod)
+}
+
+// shouldUseKeychain checks if keychain storage should be used based on the --use-keychain flag
+func shouldUseKeychain() bool {
+	useKeychain, err := profiles.GetOptionValue(options.AuthUseKeychainOption)
+	if err != nil {
+		// If we can't get the value, default to true (use keychain)
+		return true
+	}
+
+	if useKeychain == "" {
+		// If not set, default to true
+		return true
+	}
+
+	// Parse the string value to bool
+	return useKeychain == "true"
+}
+
+// SaveTokenForMethod saves an OAuth2 token to the keychain using the specified authentication method key
+// Falls back to file storage if keychain operations fail or if --use-keychain=false
+func SaveTokenForMethod(token *oauth2.Token, authMethod string) error {
+	// Check if user disabled keychain
+	if !shouldUseKeychain() {
+		// Directly save to file storage
+		return saveTokenToFile(token, authMethod)
+	}
+
+	storage := getTokenStorage(authMethod)
+
+	// Try keychain storage first
+	err := storage.SaveToken(token)
+	if err != nil {
+		// Fallback to file storage if keychain fails
```

Comment:
The documentation claims that tokens are *always* stored to the file, which is not the case in this implementation